### PR TITLE
Update docs.yaml for DOC-4040

### PR DESCRIPTION
**Pull request by emeliawilkinson24**
Added/merged the following lines with existing content per [DOC-4040](https://datastax.jira.com/browse/DOC-4040) guidelines:

checks:
  external_links:
    exclude:
      - 'https://twitter.com/dsJavaDriver'
      - 'https://twitter.com/datastaxeng' - 'https://twitter.com/datastax' - 'https://projectreactor.io' - 'https://docs.datastax.com/en/drivers/java/4.[0-9]+/com/datastax/oss/driver/internal/' - 'http://www.planetcassandra.org/blog/user-defined-functions-in-cassandra-3-0/' - 'http://www.planetcassandra.org/making-the-change-from-thrift-to-cql/' - 'https://academy.datastax.com/slack' - 'https://community.datastax.com/index.html' - 'https://micrometer.io/docs' - 'http://datastax.github.io/java-driver/features/shaded_jar/' internal_links: exclude: - 'netty_pipeline/' - '../core/' - '%5Bguava%20eviction%5D'

rewrites:
  - search: 'https://(helpdocs|docs).datastax.com/(en/astra/)?(aws|gcp)/(dscloud/apollo|doc/dscloud/astra)/dscloudGettingStarted.html' replace: 'https://docs.datastax.com/en/astra-serverless/docs/getting-started/getting-started.html'
  - search: 'https://(helpdocs|docs).datastax.com/(en/astra/)?(aws|gcp)/(dscloud/apollo|doc/dscloud/astra)/dscloudObtainingCredentials.html' replace: 'https://docs.datastax.com/en/astra-serverless/docs/connect/secure-connect-bundle.html'
  - search: 'https://(helpdocs|docs).datastax.com/(en/astra/)?(aws|gcp)/(dscloud/apollo|doc/dscloud/astra)/dscloudShareClusterDetails.html' replace: 'https://docs.datastax.com/en/astra-serverless/docs/manage/org/manage-tokens.html'
  - search: 'http://downloads.datastax.com/java-driver/' replace: 'https://downloads.datastax.com/#datastax-drivers'
  - search: 'https://docs.datastax.com/en/drivers/java/(4.[0-9]+)/com/datastax/oss/driver/api/mapper/EntityHelper.html' replace: 'https://docs.datastax.com/en/drivers/java/\1/com/datastax/oss/driver/api/mapper/entity/EntityHelper.html'
  - search: '(http|https)://www.datastax.com/drivers/java/' replace: 'https://docs.datastax.com/en/drivers/java/'
  - search: 'http://docs.datastax.com/en/drivers/java' replace: 'https://docs.datastax.com/en/drivers/java'
  - search: 'https://docs.astra.datastax.com/docs/creating-your-astra-database' replace: 'https://docs.datastax.com/en/astra-serverless/docs/getting-started/create-db-choices.html'
  - search: 'https://docs.astra.datastax.com/docs/obtaining-database-credentials' replace: 'https://docs.datastax.com/en/astra-serverless/docs/connect/secure-connect-bundle.html'
  - search: 'https://docs.datastax.com/en/drivers/java/(4.[0-9]+)/com/datastax/oss/driver/api/core/cql/SyncCqlSession.html%60' replace: 'https://docs.datastax.com/en/drivers/java/\1/com/datastax/oss/driver/api/core/cql/SyncCqlSession.html'
  - search: 'https://code.google.com/p/snappy/' replace: 'https://google.github.io/snappy/'
  - search: 'https://code.google.com/p/guava-libraries/wiki/ListenableFutureExplained' replace: 'https://github.com/google/guava/wiki/ListenableFutureExplained'
  - search: 'https://community.datastax.com/index.html' replace: 'https://www.datastax.com/workshops'

[DOC-4040]: https://datastax.jira.com/browse/DOC-4040?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs.yaml [hunk 1]
```diff
@@ -102,3 +118,30 @@ versions:
...
+  - search: 'https://docs.astra.datastax.com/docs/obtaining-database-credentials'
+    replace: 'https://docs.datastax.com/en/astra-serverless/docs/connect/secure-connect-bundle.html'
+  - search: 'https://docs.datastax.com/en/drivers/java/(4.[0-9]+)/com/datastax/oss/driver/api/core/cql/SyncCqlSession.html%60'
+    replace: 'https://docs.datastax.com/en/drivers/java/\1/com/datastax/oss/driver/api/core/cql/SyncCqlSession.html'
+  - search: 'https://code.google.com/p/snappy/'
+    replace: 'https://google.github.io/snappy/'
+  - search: 'https://code.google.com/p/guava-libraries/wiki/ListenableFutureExplained'
+    replace: 'https://github.com/google/guava/wiki/ListenableFutureExplained'
+  - search: 'https://community.datastax.com/index.html'
+    replace: 'https://www.datastax.com/workshops'
```
Comment (joao-r-reis) [inline docs.yaml:147]: There's a lot of these that are directly referencing the java driver, either they need to be modified to refer to the csharp driver or deleted.

**Review by joao-r-reis — CHANGES_REQUESTED** [review]

**Comment by jgillenwater** [comment]
@hhughes @absurdfarce What do we need to change so this PR passes the checks?

File: docs.yaml (hunk 1 above)
Comment (jgillenwater) [reply docs.yaml:147]: @emeliawilkinson24 This will apply to all of the PRs in other driver builds.

Comment (joao-r-reis) [reply docs.yaml:147]: When does it make sense to replace the link in the doc files themselves vs adding a rewrite here in the docs.yaml?

**Pull request closed without merging**